hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/lock/LeasePathLock.java

```diff
@@ -67,6 +67,16 @@ protected void acquire(TransactionLocks locks) throws IOException {
     for (Lease lease : leaseLock.getLeases()) {
       acquireLeasePaths(lease);
     }
+
+    Lease singleFileLease = leaseLock.getSingleFileLease();
+    if (singleFileLease != null){
+      setLockMode(lockType);
+      LeasePath leasePath = acquireLock(lockType, LeasePath.Finder.ByPath,
+              leaseLock.getSingleFileLock());
+      if(leasePath != null){
+        leasePaths.add(leasePath);
+      }
```

Comment:
can you do all of this in its own acquire function as we do for the other cases, it makes things easier to read.

Reply:
let me see how it looks after creating a new class

Reply:
Refactored